controller/src/beerocks/master/beerocks_master_mrouter.cpp

```
@@ -118,7 +118,12 @@ bool master_mrouter::handle_cmdu(Socket *sd, ieee1905_1::CmduMessageRx &cmdu_rx)
 {
     std::shared_ptr<sMasterSockets> soc = nullptr;
     auto uds_header                     = message_com::get_uds_header(cmdu_rx);
-    uint16_t length                     = uds_header->length;
+    if (uds_header == nullptr) {
```

More a coding style question than an actual review comment:

Somebody (@tomereli ?) said at some point in reviews that for smartptrs, you should use `if (!ptr)` rather than `if (ptr == nullptr)`. If that is the case, I think we should do the same for normal pointers.

Whoever made that comment should update CODINGSTYLE.md with that information.

---

@morantr I think you said that KW fails on such checks, and we need to use `if (!ptr)` rather than `if (ptr == nullptr)`.

---

Yes, I said it. But it only applies to smart pointers, because I remember tons of complaints about it when fixing Klocwork issues. For some reason, Klocwork didn't like.
Anyway, it does not apply to regular pointers which is the case here.

---

> Anyway, it does not apply to regular pointers which is the case here.

That's why my original comment was: "I think we should do the same for normal pointers." Unless, of course, klockwork doens't like `!ptr` for normal pointers.